streams/src/main/java/org/apache/kafka/streams/processor/ProcessorRecordContext.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor;
+
+/**
+ * An {@link RecordContext} that also can forward to children of
+ * the current {@link org.apache.kafka.streams.processor.internals.ProcessorNode}
+ */
+public interface ProcessorRecordContext extends RecordContext {
```

Comment:
I think I must have missed the motivation of creating this new interface, could you elaborate a bit? More specifically, why we could not just use `ProcessorContext.forward` that takes account the inner field `ProcessorContext`, which basically just copy-paste the logic in `ProcessorRecordContextImpl`'s functions?